### PR TITLE
Neo4jPropertyGraphStore.get() check for id prop

### PR DESCRIPTION
**Pull request by akollegger**
# Description

`Neo4jPropertyGraphStore.get()` makes assumptions about the current graph which may be wrong, particularly if an existing graph is being used which has nodes that lack an `id` property. The `get()` may construct the wrong `WHERE` clause if both `ids` and `properties` are passed in as arguments.

This fix:
- always check for existence of id property with a `WHERE e.id IS NOT NULL` clause
- allows for both ids and properties to be passed in together

Fixes # (issue)

I search, but have not found an existing issue. Shall I open one? 

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py [hunk 1]
```diff
@@ -341,19 +341,19 @@ def get(
         cypher_statement = "MATCH (e) "
 
         params = {}
-        if properties or ids:
-            cypher_statement += "WHERE "
+        # if properties or ids:
```
Comment (logan-markewich) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py:344]: Lets remove this commented code

**Review by logan-markewich — COMMENTED** [review]
makes sense. We can also bump the version on the package integraiton in the pyproject.toml so that it publishes

File: llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py (hunk 1 above)
Comment (akollegger) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py]: Done.

**Review by nerdai — APPROVED** [review]
thanks @akollegger!

**Pull request merged**